File: Week5FragmentJava/app/src/main/java/unram/psti/week5_fragment/MainActivity.java
package unram.psti.week5_fragment;

import android.support.v4.app.FragmentManager;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // instansiasi Fragment Manager
        FragmentManager fragmentManager = getSupportFragmentManager();
        HomeFragment homeFragment = new HomeFragment();

        // transaksi fragment menggunakan fragment manager
        fragmentManager
                .beginTransaction()
                .add(R.id.fragment_container, homeFragment, HomeFragment.class.getSimpleName())
                .commit();
    }
}

File: Week5FragmentKotlin/app/src/main/java/unram/psti/week5fragmentkotlin/MainActivity.kt
package unram.psti.week5fragmentkotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // menambahkan fragment pada main activity
        // instansiasi fragment manager
        val mFragmentManager = supportFragmentManager
        val mHomeFragment = HomeFragment()
        val fragment = mFragmentManager.findFragmentByTag(HomeFragment::class.java.simpleName)

        // transaski fragment
        if (fragment !is HomeFragment) {
            Log.d("FragmentApp", "Fragment Name :" + HomeFragment::class.java.simpleName)
            mFragmentManager
                .beginTransaction()
                .add(R.id.fragment_container, mHomeFragment, HomeFragment::class.java.simpleName)
                .commit()
        }
    }
}